dso-l2/src/main/java/com/tc/objectserver/handler/ProcessTransactionHandler.java

```diff
@@ -106,6 +108,14 @@ public synchronized void reconnectComplete() {
     public void handleEvent(TCMessage context) throws EventHandlerException {
       NodeID destinationID = context.getDestinationNodeID();
       invokeReturn.remove((ClientID)destinationID, context);
+      if(context instanceof VoltronEntityMultiResponse) {
+        VoltronEntityMultiResponse voltronEntityMultiResponse = (com.tc.entity.VoltronEntityMultiResponse) context;
+        for (TransactionID transactionID : voltronEntityMultiResponse.getReceivedTransactions()) {
+          waitForTransactionOrderPersistenceFuture(transactionID);
+        }
+      } else if(context instanceof VoltronEntityAppliedResponse) {
```

Comment:
I can see only 2 types from the code, so have added a assert to make sure future messages additions gets handled properly.